File: script/python/game/scene/game_actor.py

class GameActor(object):
    def __init__(self, actor_id):
        self.actor_id = actor_id

    def is_player(self):
        return False

    def is_npc(self):
        return False

    def check_can_see(self, actor):
        return True

    def on_actor_enter_sight(self, actor):
        print("on_actor_enter_sight")

    def on_actor_leave_sight(self, actor):
        print("on_actor_leave_sight")

File: script/lua/game/scene/game_actor.lua
require("base.object")
require("util.logger")
require("util.multi_index_elem")

clsGameActor = clsMultiIndexElem:Inherit("clsGameActor")

function clsGameActor:__init__(game_scene, actor_id, index_attr_names)
    Super(clsGameActor).__init__(self, index_attr_names)
    self.actor_id = actor_id
    self.game_scene = game_scene
end

function clsGameActor:is_player()
    return false
end

function clsGameActor:is_npc()
    return false
end

function clsGameActor:check_can_see(actor)
    return true
end
